hydrolib/core/utils.py

```diff
@@ -175,3 +176,127 @@ def get_operating_system() -> OperatingSystem:
+        return PathStyle.WINDOWSLIKE
+    else:
+        return PathStyle.UNIXLIKE
+
+
+class FilePathStyleConverter:
+    """Class for converting file paths between different path styles."""
+
+    def __init__(self) -> None:
+        """Initializes a new instance of the FilePathStyleConverter class."""
+        self._os_path_style = get_path_style_for_current_operating_system()
+
+    def convert_to_os_style(self, file_path: Path, source_path_style: PathStyle) -> str:
+        """Convert the file path from the source path style to the path style of the current operating system.
+
+        Args:
+            file_path (Path): The file path to convert to the OS path style.
+            file_path_style (PathStyle): The file path style of the given file path.
+
+        Returns:
+            str: The converted file path with OS path style.
+
+        Raises:
+            NotImplementedError: When this function is called with a PathStyle other than WINDOWSLIKE or UNIXLIKE.
+        """
+
+        return FilePathStyleConverter._convert(
+            file_path, source_path_style, self._os_path_style
+        )
+
+    def convert_from_os_style(
+        self, file_path: Path, target_path_style: PathStyle
+    ) -> str:
+        """Convert the file path from the path style of the current operating system to the target path style.
+
+        Args:
+            file_path (Path): The file path to convert to the target path style.
+            target_path_style (PathStyle): The target path style to convert the file path to.
+
+        Returns:
+            str: The converted file path with the target path style.
+
+        Raises:
+            NotImplementedError: When this function is called with a PathStyle other than WINDOWSLIKE or UNIXLIKE.
+        """
+
+        return FilePathStyleConverter._convert(
+            file_path, self._os_path_style, target_path_style
+        )
+
+    @classmethod
+    def _convert(
+        cls, file_path: Path, source_path_style: PathStyle, target_path_style: PathStyle
+    ) -> str:
+        if source_path_style == target_path_style:
+            return file_path.as_posix()
+
+        if (
+            source_path_style == PathStyle.UNIXLIKE
+            and target_path_style == PathStyle.WINDOWSLIKE
+        ):
+            return FilePathStyleConverter._from_posix_to_windows_path(file_path)
+        elif (
+            source_path_style == PathStyle.WINDOWSLIKE
+            and target_path_style == PathStyle.UNIXLIKE
+        ):
+            return FilePathStyleConverter._from_windows_to_posix_path(file_path)
+        else:
+            raise NotImplementedError(
+                f"Cannot convert {source_path_style} to {target_path_style}"
+            )
+
+    @classmethod
+    def _from_posix_to_windows_path(cls, posix_path: Path) -> str:
+        is_relative = not posix_path.as_posix().startswith("/")
+
+        if is_relative:
+            return posix_path.as_posix()
+
+        root = posix_path.parts[1]
```

Comment:
I think we should not do this automatically, there is no guarantee that a Unix-like drive `/p/` is on Windows systems available with the same root-name `P:/`. Or at least allow the user to supply a drive-letter dictionary. Can we add this to the `ModelSaveSettings` class?

For example: a Windows `T:\` could map on some Linux system to `/store/transfer/`.
Maybe something for a followup issue.

Reply:
Added #480